unity/Holo/Assets/Editor/ModelImport/FileSeriesImporter.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+namespace ModelImport
+{    
+    //A class for importing a file series from a directory.
+    public class FileSeriesImporter
+    {
+        public GameObject ModelGameObject { get; private set; }
+        public Mesh ModelMesh { get; private set; }
+
+        public GameObject[] gameObjectArray;
+        public Mesh[] meshArray;
+        private string[] filePaths;
+        private string extension;
+        private FileImporter fileImporter;
+        private Dictionary<string, Vector3> boundingVertices = new Dictionary<string, Vector3>();
+        private bool dataflow = false;
+        
+        // Object constructor, initiates file series import.
+        public void ImportData(string fileSeriesDirectory, string gameObjectName)
+        {
+            ModelGameObject = new GameObject
+            {
+                name = gameObjectName
+            };
+            ModelMesh = new Mesh();
+            if (gameObjectName.Contains("dataflow"))
+            {
+                dataflow = true;
+            }
+            GetFilepaths(fileSeriesDirectory);
+            ImportFiles();
+            ConfigureMesh();
+        }
+
+        //Gets filepaths of particular frames and their extension
+        private void GetFilepaths(string rootDirectory)
+        {
+            filePaths = Directory.GetFiles(rootDirectory + @"\");
+            extension = Path.GetExtension(filePaths[0]);
+        }
+
+        //Loads meshes from separate files into Mesh Object as BlendShapeFrames
+        private void ImportFiles()
+        {
+            fileImporter = new FileImporter(extension, dataflow);
+
+            //Configuring progress bar
+            float progressChunk = (float) 1 / filePaths.Length;
+            bool cancelImport = EditorUtility.DisplayCancelableProgressBar("Conversion in progress", "Converting file nr: " + 0.ToString(), 0);
+
+            bool firstMesh = true;
+            for(int i = 0; i < filePaths.Length; i++)
+            {
+                if (cancelImport)
+                {
+                    AbortImport();

```

Comment:
`return` would be nicer with ErrorLog instead of exception.

Reply:
I use throw exception here, because it stops method from running and is converted into ErrorLog output anyway, while ErrorLog will keep function running.